distribution/lib/Standard/Base/0.0.0-dev/src/Random.enso

```diff
@@ -1,99 +1,307 @@
+       - end_time: The upper bound of the range to pick from (inclusive).
+
+       The generated time is at a granularity of 1 second.
+
+       Note: the time range end is inclusive because it is impossible to specify
+       an end time after the last second of the day.
+
+       > Example
+         Generate a random time.
+
+             import Standard.Base.Random.Random
+
+             start = Time_Of_Day.new 8 12 15
+             end = Time_Of_Day.new 9 40 2
+             t = Random.time start end
+    time : Time_Of_Day -> Time_Of_Day -> Time_Of_Day
+    time (start_time:Time_Of_Day=(Time_Of_Day.new 0 0 0)) (end_time:Time_Of_Day=(Time_Of_Day.new 23 59 59)) =
+        Random.default_instance.time start_time end_time
+
+    ## GROUP Random
+       Return a new UUID.
+
+       ! `uuid` does not use the underlying `java.util.Random` instance, and is
+       therefore not affected by the seed.
+
+       > Example
+         Generate a uuid.
+
+             import Standard.Base.Random.Random
+
+             uuid = Random.uuid
+    uuid : Text
+    uuid = Random.default_instance.uuid
+
+    ## GROUP Random
+       Returns a new `Vector` containing a random sample of the input vector,
+       with or without replacement.
+
+       Arguments:
+       - vector: The `Vector` to take items from.
+       - count: The number of items to return.
+       - with_replacement: If false, each value in the `Vector` can only be
+         selected once during a single call; the returned values will contain no
+         duplicates. If true, values from the `Vector` can be returned multiple
+         times and duplicates are possible.
+
+       If `with_replacement` is false, and `count` is equal to or greater than
+       the size of the collection, a random permutation of the collection is
+       returned.
+
+       > Example
+         Sample a vector.
+
+             import Standard.Base.Random.Random
+
+             items = Random.items [1, 2, 3, 4]
+    item : Vector Any -> Integer -> Boolean -> Any
+    item (v:Vector) (count:Integer=1) (with_replacement:Boolean=False) = Random.default_instance.item v count with_replacement
+
+    ## GROUP Random
+       Returns a random permutation of the input `Vector`.
+
+       Arguments:
+       - vector: The `Vector` to permute.
+
+       > Example
+         Permute a vector.
+
+             import Standard.Base.Random.Random
+
+             permuted = Random.permute [1, 2, 3, 4]
+    permute : Vector Any -> Vector Any
+    permute (v:Vector) = Random.default_instance.permute v
 
     ## PRIVATE
-       Serialise to JS_Object
-    to_js_object : JS_Object
-    to_js_object self =
-        JS_Object.from_pairs [["type", "Random_Number_Generator"], ["constructor", "new"]]
+       Return the default instance.
+    default_instance : Random_Generator
+    default_instance = Random_Generator.Value Random_Utils.getRandomSingleton
 
+# PRIVATE
+type Random_Generator
```

Comment:
Why don't we use @Akirathan's 
- #7840

e.g. why  doesn't this "private type" (if it is really _private_) get moved into a module which starts with `private` keyword?

Reply:
> I just tried moving `Random_Generator` to a private module, but this means that I can't refer to it in `Random_Spec.enso`. This is currently truly a private type, but I still need to run tests against it.

This is my bad. I didn't implement `--disable-private-check` as mentioned in the [docs](https://github.com/enso-org/enso/pull/7840/files#diff-6cf8b56703980950b1660ca6fc2619df6336b359d463f023105a3089ed1345deR170). Sorry for that. Still, `--disable-private-check` is not yet ready and shall be ready in the follow-up PR for https://github.com/enso-org/enso/issues/8100

Reply:
The missing `--disable-private-check` option is blocking acceptance of #7840. I don't think it is dependent on #8100. As such I have just reported new issue 
- #8140